tensorflow_encrypted/protocol/pond.py

```diff
@@ -187,21 +186,25 @@ def define_private_variable(
 
     def define_public_input(
         self,
-        provider: InputProvider,
+        player: Union[str, Player],
+        data_fn: Callable,
```

Comment:
Think it's worth high lighting that it's a Callable that returns either a single tf.Tensor or list of tf.Tensors

Reply:
see naming comments elsewhere, same applies here